cylc/flow/config.py

```diff
@@ -218,10 +218,12 @@ def __init__(
         self.mem_log("config.py: before RawSuiteConfig init")
         if output_fname:
             output_fname = os.path.expandvars(output_fname)
+        # Copy template_vars to prevent addition of non serializable template
+        # vars from file being added by parsec.
         self.pcfg = RawSuiteConfig(
             fpath,
             output_fname,
-            template_vars
+            copy(template_vars)
```

Comment:
Any reason you prefer this copy here instead of in `read_and_proc`, where you modify the input `template_vars`?

Reply:
I'd rather not pass the original across at all. Is there a reason to do it the other way around?

Reply:
Well, just a matter of style I suppose. I've always considered mutating input arguments in a function with explicit return value a bit evil.

Reply:
It turned out to be a mistake anyway - I've moved the copying inside the jinja2/empy processing sections so they cannot accidentally contaminate each other.